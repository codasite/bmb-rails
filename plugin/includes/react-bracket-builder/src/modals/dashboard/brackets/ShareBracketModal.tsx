import { ModalHeader } from '../../ModalHeader'
import { CancelButton, ConfirmButton } from '../../ModalButtons'
import { useState } from 'react'
import { Modal } from '../../Modal'
import addClickHandlers from '../../addClickHandlers'
import { ReactComponent as LinkIcon } from '../../../brackets/shared/assets/link.svg'

export default function ShareBracketModal() {
  const [show, setShow] = useState(false)
  const [playBracketUrl, setPlayBracketUrl] = useState('')
  addClickHandlers({
    buttonClassName: 'wpbb-share-bracket-button',
    onButtonClick: (b) => {
      setPlayBracketUrl(b.dataset.playBracketUrl)
      setShow(true)
    },
  })
  return (
    <Modal show={show} setShow={setShow}>
      <ModalHeader text={'Share Bracket'} />
      <div className="tw-flex tw-flex-col tw-gap-10">
        <ConfirmButton
          disabled={false}
          onClick={() => {
            navigator.clipboard.writeText(playBracketUrl).catch((err) => {
              console.error(err)
              console.error(
                'If "navigator.clipboard is undefined", you may not be using a secure origin.'
              )
            })
            setShow(false)
          }}
        >
          <LinkIcon />
          <span>Copy link</span>
        </ConfirmButton>
        <CancelButton onClick={() => setShow(false)} />
      </div>
    </Modal>
  )
}
