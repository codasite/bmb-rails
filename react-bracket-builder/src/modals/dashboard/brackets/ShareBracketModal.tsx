import { ModalHeader } from '../../ModalHeader'
import { CancelButton, ConfirmButton, Link } from '../../ModalButtons'
// DO NOT REMOVE REACT IMPORT. Needed for image generator
import React from 'react'
import { Modal } from '../../Modal'
import { ReactComponent as LinkIcon } from '../../../brackets/shared/assets/link.svg'
import { ReactComponent as XLogo } from '../../../brackets/shared/assets/x-logo.svg'
import { ReactComponent as FacebookLogo } from '../../../brackets/shared/assets/facebook-logo.svg'
import { BracketData } from './BracketData'

export const FacebookShareLink = (props: { playBracketUrl: string }) => {
  const facebookShareUrl = `https://www.facebook.com/sharer/sharer.php?u=${props.playBracketUrl}`
  return (
    <Link href={facebookShareUrl} openInNewTab={true}>
      <FacebookLogo />
    </Link>
  )
}

export const TwitterShareLink = (props: {
  playBracketUrl: string
  bracketTitle: string
}) => {
  const shareText = `Play my bracket, ${props.bracketTitle?.toUpperCase()}!`
  const twitterShareUrl = `https://twitter.com/intent/tweet?text=${shareText}&url=${props.playBracketUrl}&via=BackMyBracket`
  return (
    <Link href={twitterShareUrl} openInNewTab={true}>
      <XLogo />
    </Link>
  )
}

export const CopyLinkButton = (props: {
  playBracketUrl: string
  onClick: () => void
}) => {
  return (
    <ConfirmButton
      disabled={false}
      onClick={() => {
        navigator.clipboard.writeText(props.playBracketUrl).catch((err) => {
          console.error(err)
          console.error(
            'If "navigator.clipboard is undefined", you may not be using a secure origin.'
          )
        })
        props.onClick()
      }}
    >
      <LinkIcon />
      <span>Copy link</span>
    </ConfirmButton>
  )
}

export const ShareBracketModal = (props: {
  show: boolean
  setShow: (show: boolean) => void
  bracketData: BracketData
}) => {
  console.log('ShareBracketModal', props.bracketData)
  return (
    <Modal show={props.show} setShow={props.setShow}>
      <ModalHeader text={'Share Bracket'} />
      <div className="tw-flex tw-flex-col tw-gap-10">
        <div className="tw-flex tw-gap-10 tw-items-center">
          <FacebookShareLink
            playBracketUrl={props.bracketData.playBracketUrl}
          />
          <TwitterShareLink
            playBracketUrl={props.bracketData.playBracketUrl}
            bracketTitle={props.bracketData.title}
          />
        </div>
        <CopyLinkButton
          playBracketUrl={props.bracketData.playBracketUrl}
          onClick={() => props.setShow(false)}
        />
        <CancelButton onClick={() => props.setShow(false)} />
      </div>
    </Modal>
  )
}
