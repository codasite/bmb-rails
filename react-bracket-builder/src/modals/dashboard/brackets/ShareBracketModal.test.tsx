import { render, screen } from '@testing-library/react'
import ShareBracketModal from './ShareBracketModal'
import userEvent from '@testing-library/user-event'

describe('ShareBracketModal', () => {
  test('renders modal correctly', async () => {
    const { asFragment } = render(
      <>
        <button
          className={'wpbb-share-bracket-button'}
          data-bracket-title={'Test bracket'}
          data-play-bracket-url={'testurl'}
        >
          Share Bracket
        </button>
        <ShareBracketModal />
      </>
    )
    await userEvent.click(screen.getByText('Share Bracket'))
    expect(asFragment()).toMatchSnapshot()
  })
})
