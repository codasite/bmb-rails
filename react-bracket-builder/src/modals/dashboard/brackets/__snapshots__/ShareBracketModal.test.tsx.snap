// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareBracketModal renders modal correctly 1`] = `
<DocumentFragment>
  <button
    class="wpbb-share-bracket-button"
    data-bracket-title="Test bracket"
    data-play-bracket-url="testurl"
  >
    Share Bracket
  </button>
  <div
    class="tw-fixed tw-bg-black/50 tw-top-0 tw-left-0 tw-right-0 tw-z-50 tw-w-full tw-p-4 tw-overflow-x-hidden tw-overflow-y-auto md:tw-inset-0 tw-h-[calc(100%-1rem)] tw-max-h-full tw-justify-center tw-items-center tw-flex"
    tabindex="-1"
  >
    <div
      class="tw-relative tw-max-w-[606px] tw-max-h-full tw-p-30 sm:tw-p-60 tw-rounded-16 tw-bg-dark-blue tw-grow tw-mx-20"
    >
      <h1
        class="tw-text-24 sm:tw-text-32 tw-leading-10 tw-text-white tw-uppercase tw-whitespace-pre-line tw-mb-30 tw-w-full tw-text-center tw-font-sans"
      >
        Share Bracket
      </h1>
      <div
        class="tw-flex tw-flex-col tw-gap-10"
      >
        <div
          class="tw-flex tw-gap-10 tw-items-center"
        >
          <a
            class="tw-bg-white/15 tw-flex tw-gap-16 tw-items-center tw-justify-center tw-rounded-8 tw-p-12 hover:tw-text-white/75 tw-font-sans tw-text-white tw-uppercase tw-w-full tw-text-16 tw-font-500 tw-cursor-pointer tw-border-solid tw-border tw-border-white"
            href="https://www.facebook.com/sharer/sharer.php?u=testurl"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg />
          </a>
          <a
            class="tw-bg-white/15 tw-flex tw-gap-16 tw-items-center tw-justify-center tw-rounded-8 tw-p-12 hover:tw-text-white/75 tw-font-sans tw-text-white tw-uppercase tw-w-full tw-text-16 tw-font-500 tw-cursor-pointer tw-border-solid tw-border tw-border-white"
            href="https://twitter.com/intent/tweet?text=Play my bracket, TEST BRACKET!&url=testurl&via=BackMyBracket"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg />
          </a>
        </div>
        <button
          class="tw-flex tw-flex-row tw-items-center tw-justify-center tw-font-sans tw-uppercase tw-whitespace-nowrap tw-cursor-pointer disabled:tw-cursor-default tw-leading-none tw-bg-green/15 tw-gap-16 tw-rounded-8 tw-p-12 tw-border-1 tw-border-solid tw-border-green hover:tw-text-white/75 tw-text-white tw-w-full tw-text-16 tw-font-700 disabled:tw-bg-transparent disabled:tw-text-white/20 disabled:tw-border-white/20"
        >
          <svg />
          <span>
            Copy link
          </span>
        </button>
        <button
          class="tw-flex tw-flex-row tw-items-center tw-justify-center tw-font-sans tw-uppercase tw-whitespace-nowrap tw-cursor-pointer disabled:tw-cursor-default tw-leading-none tw-bg-white/15 tw-gap-16 tw-rounded-8 tw-p-12 tw-border-none hover:tw-text-white/75 tw-text-white tw-w-full tw-text-16 tw-font-500 disabled:tw-bg-white/15 disabled:tw-text-white/50"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
